sktime/clustering/partitioning/_lloyds.py

```diff
@@ -240,6 +240,11 @@ def __init__(
 
         self._random_state = None
         self._init_algorithm = None
+        self._initial_cluster_centers = None
+
+        if isinstance(self.init_algorithm, np.ndarray):
+            self._initial_cluster_centers = self.init_algorithm
+            self.init_algorithm = lambda *args, **kwargs: self._initial_cluster_centers
```

Comment:
I could use some feedback on whether there is a better way than using this lambda function to return the initial cluster centers.

I cannot use a class method called something like `return_initial_cluster_centers` because that causes a recursion error. For example, suppose this class has the method
```python
def return_initial_cluster_centers(self, *args, **kwargs):
      return self._initial_cluster_centers
```
And the lambda function is replaced with the function above. Then, I get the following reccursion error.
```python
>>> from sktime.clustering.k_means import TimeSeriesKMeans
>>> import numpy array as np
>>> model = TimerSeriesKMeans(init_algorithm=np.array([0, 0]))
>>> model
  File "/home/alex/documents/sktime/.venv/lib/python3.10/site-packages/skbase/base/_pretty_printing/_pprint.py", line 389, in _safe_repr
    params = _changed_params(obj)
  File "/home/alex/documents/sktime/.venv/lib/python3.10/site-packages/skbase/base/_pretty_printing/_pprint.py", line 34, in _changed_params
    params = base_object.get_params(deep=False)
  File "/home/alex/documents/sktime/.venv/lib/python3.10/site-packages/skbase/base/_base.py", line 359, in get_params
    params = {key: getattr(self, key) for key in self.get_param_names()}
  File "/home/alex/documents/sktime/.venv/lib/python3.10/site-packages/skbase/base/_base.py", line 311, in get_param_names
    parameters = cls._get_init_signature()
  File "/home/alex/documents/sktime/.venv/lib/python3.10/site-packages/skbase/base/_base.py", line 283, in _get_init_signature
    init_signature = inspect.signature(init)
  File "/usr/lib/python3.10/inspect.py", line 3247, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped,
  File "/usr/lib/python3.10/inspect.py", line 2995, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
  File "/usr/lib/python3.10/inspect.py", line 2383, in _signature_from_callable
    _get_signature_of = functools.partial(_signature_from_callable,
RecursionError: maximum recursion depth exceeded while calling a Python object
```
I am guessing this is something to do with the `BaseClusterer` class but I am not sure exactly.

Reply:
how about you skip the idea of passing a function entirely - if `init_algorithm` is `numpy.array`, in `_check_params`, just get the array, instead of calling a function.

Reply:
Switched to avoiding passing around function in commit 109bbc3.